packages/memsocket/memtest.nix

```diff
@@ -0,0 +1,39 @@
+# Copyright 2022-2023 TII (SSRC) and the Ghaf contributors
+# SPDX-License-Identifier: Apache-2.0
+{
+  stdenv,
+  lib,
+  gcc,
+  gnumake,
+  fetchFromGitHub,
+  ...
+}:
+stdenv.mkDerivation {
+  name = "memtest";
+
+  src = fetchFromGitHub {
+    owner = "tiiuae";
+    repo = "shmsockproxy";
+    rev = "4eef1a7e1f6993a9ba0ef1aa066145c410e2eec6";
+    sha256 = "sha256-uaKBtb+bD42RK/Z96dhMrGxNXg+ZV/6iW52ouNQ+QSY=";
+  };
+
+  nativeBuildInputs = [gcc gnumake];
```

Comment:
GCC and gnumake is in stdenv, no need to set them explicitly. Also I suggest to join memsocket and memtest to single package (just add there root makefile to make/make install all at once.

Reply:
I decided to remove the memtest component. It's used only for checking the speed and reliability of the socket-to-socket connection.